### ISSUE TITLE
M2 and Vias to M2

### ISSUE THREAD

**Issue by blu006** [tkimva/ucr-eecs168#16]
Hello,

Can we use the `M2` layer?

Which vias do we use to the `M2` layer?  `VIA1` layer?

- Brandon

**Comment by tkimva** [comment]
In order to connect different layers to each other, all metal layers and contacts/VIAs must exist in between them. For example, in order to connect a metal 2 layer (M2) to diffusion (DIFF), one contact layer (CO), metal 1 layer (M1), and one VIA1 layer must exist between them. If you miss any layer, then it will not be passed in LVS.

If you're still confused, I will write an small tutorial today soon. please hold.

**Comment by blu006** [comment]
Ahh, that makes sense.

So,
`CO` is between `DIFF` and `M1`,
`VIA1` is between `M1` and `M2`?

Thank you for your help!

**Comment by tkimva** [comment]
I made and draw quick tutorial for M2 layer use. This has been added to lab3 tutorial as well.

-Extra tutorial for use multiple layer

Using M2 Layer (or higher like M3, M4...) in Layout
For designs that require require an extra metal layers in layout, designers can use a metal layer higher up (like M2) to make connections if the lower metal layers (like M1) are too constricting to allow any other connections. In the ring oscillator layout, M1 (blue layer) is replaced with M2 (pink layer). See Fig 107 for reference (Ring Oscillator with M2 layer). Also note that M1 layers can run under M2 layers without physically connecting unless there is a VIA1 layer in between them.

![fig_107](https://cloud.githubusercontent.com/assets/993376/12764915/f6c75ad0-c9b0-11e5-9e8d-b59c329bf018.png)


fig107

Fig. 107. Multiple layer (M2) example

In order to connect different layers to each other, all metal layers and contacts/VIAs must exist in between them. For example, to connect a metal 2 layer (M2) to diffusion (DIFF), a contact layer (CO), metal 1 layer (M1), and a VIA1 layer must exist between them. See figure 107 for reference. Also see figure 106 in circle A for a layout drawing example. To use higher metal layers not shown, use their corresponding VIA layers to interconnect between the desired layers. For example, to connect M2 to M3, a VIA2 layer/contact must be drawn in between them.

Contact (CO) size 0.13um x 0.13um but VIA1 size is 0.14um x 0.14um for 90nm design.
